src/textual/widgets/_tree.py

```diff
@@ -513,6 +513,17 @@ def get_node_at_line(self, line_no: int) -> TreeNode[TreeDataType] | None:
         else:
             return line.node
 
+    def get_node_by_id(self, node_id: NodeID) -> TreeNode[TreeDataType]:
```

Comment:
I think this might deserve a custom exception. A `KeyError` would too easily be confused with another logic error. We would also want the opportunity to add a custom message like "No tree node with id 'foo'".

Let's also add a `Raises:` section in the docstring.

Reply:
Sold!

Reply:
Done: 728acc27c0d90a9e00184871a78df840d1bf69d3